### PR TITLE
test: unit test Web3FunctionBuilder

### PR DESCRIPTION
**Pull request by mkykadir**

### CODE REVIEW

File: src/lib/builder/__test__/valid-schema/index.js
Comment (brandonchuah) [inline src/lib/builder/__test__/valid-schema/index.js:1]: Should we git ignore this file here?

File: src/lib/builder/__test__/valid-schema/source.js
[the remark this answers is not in the document]
Comment (mkykadir) [reply src/lib/builder/__test__/valid-schema/source.js:1]: I wanted to keep them as the builder requires those files and they're part of the test event their content is not important

File: src/lib/builder/Web3FunctionBuilder.spec.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  test("should fail when input path does not exist", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("not-existing")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes("Missing Web3Function schema")
+      ).toBeTruthy();
```
Comment (goums) [inline src/lib/builder/Web3FunctionBuilder.spec.ts:27]: why not use this more compact matcher instead:
```suggestion
    expect(res.error.message).toMatch("Missing Web3Function schema");
```

File: src/lib/builder/Web3FunctionBuilder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+  test("should fail when schema userArgs include unknown types", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("invalid-schema-timeout")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(res.error.message.includes("'timeout' must be")).toBeTruthy();
+    }
```
Comment (goums) [inline src/lib/builder/Web3FunctionBuilder.spec.ts:119]: I think this one is a copy/paste of the above and is not actually checking the user args

**Review by goums — CHANGES_REQUESTED** [review]

File: src/lib/builder/Web3FunctionBuilder.spec.ts (hunk 2 above)
Comment (mkykadir) [reply src/lib/builder/Web3FunctionBuilder.spec.ts]: nice catch, fixed in 5c93b2f081209d252d59f59f2d8c50cede106f04

File: src/lib/builder/Web3FunctionBuilder.spec.ts (hunk 1 above)
Comment (mkykadir) [reply src/lib/builder/Web3FunctionBuilder.spec.ts]: I thought is was checking for exact match, 5a3b57652d8bc12a6d4f849b1f72c2fcb8ac7eee

**Review by goums — APPROVED** [review]

**Pull request merged**